pyproject.toml

```diff
@@ -22,8 +22,14 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.8"
-
-jsonschema = "*"
+# Pinning jsonschema because its dependencies introduced
+# a dependency of rustc, which we don't want to have installed
+# in charms that use SDI
+# For more information go to canonical/bundle-kubeflow/issues/648
+# For more reference go to:
+# https://github.com/python-jsonschema/jsonschema/issues/1117
+# https://github.com/python-jsonschema/jsonschema/issues/1114
+jsonschema = "^4.17.3"
```

Comment:
`^4.17.3` pins to a [semver compatible version](https://python-poetry.org/docs/dependency-specification/#caret-requirements), so it is equivalent to `>4.17.3, <5`.  It doesn't pin away from 4.18 like we're trying

This should work instead:
```suggestion
# Pinning jsonschema because its dependencies introduced
# a dependency of rustc, which we don't want to have installed
# in charms that use SDI
# For more information go to canonical/bundle-kubeflow/issues/648
# For more reference go to:
# https://github.com/python-jsonschema/jsonschema/issues/1117
# https://github.com/python-jsonschema/jsonschema/issues/1114
jsonschema = ">4, <4.18"
```

I've run the unit tests with `jsonschema==4.0.0` and it worked, so that gives a little flexibility but not too much.  what do you think?

Reply:
@addyess you beat me to it! :D

Reply:
oops, sorry